### PR TITLE
Add String support

### PR DESCRIPTION
**Pull request by ikesyo**
Could you please review this?

---

string_tweak

### CODE REVIEW

**Comment by ikesyo** [comment]
ping @bryanjclark?

**Comment by bryanjclark** [comment]
Hey! Thanks for the diff!

I'll take a look at this soon. Since it's a new `TweakOptionType`, it means that the diff is large - with interaction design changes, etc - so it might be a while until I get it landed!

Thanks for doing this!

File: iOS Example/iOS Example.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -85,6 +85,7 @@
 				93A3AEFC1BF142A400CAD43B /* Products */,
 			);
 			sourceTree = "<group>";
+			usesTabs = 1;
```
Comment (nixterrimus) [inline iOS Example/iOS Example.xcodeproj/project.pbxproj]: You probably didn't mean to commit this.  Can you remove it?

Comment (ikesyo) [inline iOS Example/iOS Example.xcodeproj/project.pbxproj]: No, this is intentional. The project aparently uses tab indentation but was not configured as such. This prevents commiting wrong indentation (as seen in the diff).

**Comment by bryanjclark** [comment]
Hey! @nixterrimus and I are going to start reviewing stuff here weekly for a while; we'll be tackling this one next week! 😄

File: iOS Example/iOS Example.xcodeproj/project.pbxproj (hunk 1 above)
Comment (nixterrimus) [inline iOS Example/iOS Example.xcodeproj/project.pbxproj]: This would be a super easy commit to land all but itself.  Would you mind moving this to another PR?

Comment (ikesyo) [inline iOS Example/iOS Example.xcodeproj/project.pbxproj]: Just reverted 6c96efa

**Comment by algal** [comment]
We are avid Tweaks users at Topology Eyewear and would love to use this feature. Please let me know if there’s some way we could help with dev or testing of this, if that would be helpful.

**Comment by ekimia** [comment]
++++1

**Comment by ikesyo** [comment]
Ping @bryanjclark?

**Comment by ikesyo** [comment]
Ping @nixterrimus and @bryanjclark?

**Comment by ikesyo** [comment]
Ping @nixterrimus and @bryanjclark?

**Comment by bryanjclark** [comment]
Hi @ikesyo, yes, I see your pings, and I understand you're eager to see this land in the main repository, but we haven't had time to review this yet as our team is busy with other things. We'll get to it eventually — in the meantime, thank you for your patience. Cheers.

**Comment by JoeSzymanski** [comment]
Adding a +1 to this. We have some work that would benefit from being able to change string values.

**Comment by callumboddy** [comment]
+1

**Comment by dannyhertz** [comment]
+1

**Comment by ikesyo** [comment]
Hi @bryanjclark, looks like finally you are re-working on this project now. So could you please review this long-standing pull request? This should be straightforward implementation so will not be difficult to review I think.

File: README.md [hunk 2]
```diff
@@ -191,7 +193,7 @@ public struct FirstTweaksLibrary: TweakLibraryType {
 ```
 
 #### Why can’t any type be used for a `Tweak`?
-While `Tweak<T>` is generic, we have to restrict `T` to be `TweakableType` so that we can guarantee that each kind of `T` can be represented in our editing interface and persisted on disk. More types would be awesome, though! It’d be neat to support dictionaries, closures, strings, and other things.
```
Comment (bryanjclark) [inline README.md]: Thanks for updating this!! 😄

File: SwiftTweaks/TweakTableCell.swift [hunk 3]
```diff
@@ -166,6 +167,17 @@ internal final class TweakTableCell: UITableViewCell {
 			let accessoryFrame = colorControlFrame.union(textFrame).union(disclosureArrowFrame)
 			accessory.bounds = accessoryFrame.integral
 
+		case .string:
+			let textFrame = CGRect(
+				origin: .zero,
+				size: CGSize(
+					width: bounds.width * TweakTableCell.stringTextWidthFraction,
```
Comment (bryanjclark) [inline SwiftTweaks/TweakTableCell.swift]: Hyper-nit: let’s round this to the nearest whole pixel. (I’ll do this in a follow-up diff)

File: iOS Example/iOS Example/ExampleTweaks.swift [hunk 4]
```diff
@@ -27,7 +27,8 @@ public struct ExampleTweaks: TweakLibraryType {
 	public static let colorText1 = Tweak("Text", "Color", "text-1", UIColor.black)
 	public static let colorText2 = Tweak("Text", "Color", "text-2", UIColor(hue: 213/255, saturation: 0.07, brightness: 0.58, alpha: 1))
 
-    public static let title = Tweak<StringOption>.stringList("Text", "Text", "Title", options: ["SwiftTweaks", "Welcome!", "Example"])
+	public static let title = Tweak<StringOption>.stringList("Text", "Text", "Title", options: ["SwiftTweaks", "Welcome!", "Example"])
```
Comment (bryanjclark) [inline iOS Example/iOS Example/ExampleTweaks.swift]: Maybe let’s do `Title (StringList)` and `Subtitle (String)` for demo purposes! (I’ll do this in a follow-up diff)

**Comment by ikesyo** [comment]
@bryanjclark Thanks for the review and merging! 💖
